treeherder/webapp/api/logslice.py

```diff
@@ -79,8 +83,7 @@ def list(self, request, project, jm):
                 return Response(lines)
 
             except Exception as e:
-                logging.error(e)
-                raise ResourceNotFoundException("log file not found")
+                raise
```

Comment:
Catching and then raising without doing anything else should be the same as not having the except - we can just have the try-finally, right?

That said, I think we still need to handle the file is missing case - this is common, since logs are deleted from FTP after 30 days. The UI currently looks for the 404 and displays a "log not found". Seems like we just need to wrap the get_log_handle() and try-except that for HTTPError of type 404.

Happy to break this part out into another PR if it unblocks landing the Django 1.7 compat fix :-)